src/SBSpergel.cpp

```diff
@@ -0,0 +1,715 @@
+                                            double y0, double dy, int iy_zero) const
+    {
+        dbg<<"SBSpergel fillKValue\n";
+        dbg<<"x = "<<x0<<" + ix * "<<dx<<", ix_zero = "<<ix_zero<<std::endl;
+        dbg<<"y = "<<y0<<" + iy * "<<dy<<", iy_zero = "<<iy_zero<<std::endl;
+        if (ix_zero != 0 || iy_zero != 0) {
+            xdbg<<"Use Quadrant\n";
+            fillKValueQuadrant(val,x0,dx,ix_zero,y0,dy,iy_zero);
+        } else {
+            xdbg<<"Non-Quadrant\n";
+            assert(val.stepi() == 1);
+            const int m = val.colsize();
+            const int n = val.rowsize();
+            typedef tmv::VIt<std::complex<double>,1,tmv::NonConj> It;
+
+            x0 *= _r0;
+            dx *= _r0;
+            y0 *= _r0;
+            dy *= _r0;
+
+            for (int j=0;j<n;++j,y0+=dy) {
+                double x = x0;
+                double ysq = y0*y0;
+                It valit = val.col(j).begin();
+                for (int i=0;i<m;++i,x+=dx) {
+                    double ksq = x*x + ysq;
+                    *valit++ = _flux * _info->kValue(ksq);
+                }
+            }
+        }
+    }
+
+    void SBSpergel::SBSpergelImpl::fillXValue(tmv::MatrixView<double> val,
+                                            double x0, double dx, double dxy,
+                                            double y0, double dy, double dyx) const
+    {
+        dbg<<"SBSpergel fillXValue\n";
+        dbg<<"x = "<<x0<<" + ix * "<<dx<<" + iy * "<<dxy<<std::endl;
+        dbg<<"y = "<<y0<<" + ix * "<<dyx<<" + iy * "<<dy<<std::endl;
+        assert(val.stepi() == 1);
+        assert(val.canLinearize());
+        const int m = val.colsize();
+        const int n = val.rowsize();
+        typedef tmv::VIt<double,1,tmv::NonConj> It;
+
+        x0 *= _inv_r0;
+        dx *= _inv_r0;
+        dxy *= _inv_r0;
+        y0 *= _inv_r0;
+        dy *= _inv_r0;
+        dyx *= _inv_r0;
+
+        double x00 = x0; // Preserve the originals for below.
+        double y00 = y0;
+        It valit = val.linearView().begin();
+        for (int j=0;j<n;++j,x0+=dxy,y0+=dy) {
+            double x = x0;
+            double y = y0;
+            for (int i=0;i<m;++i,x+=dx,y+=dyx) {
+                double r = sqrt(x*x + y*y);
+                *valit++ = _xnorm * _info->xValue(r);
+            }
+        }
+
+        // Check for (0,0) in disguise as in Sersic
+        double det = dx * dy - dxy * dyx;
+        double i0 = (-dy * x00 + dxy * y00) / det;
+        double j0 = (dyx * x00 - dx * y00) / det;
+        dbg<<"i0, j0 = "<<i0<<','<<j0<<std::endl;
+        dbg<<"x0 + dx i + dxy j = "<<x00+dx*i0+dxy*j0<<std::endl;
+        dbg<<"y0 + dyx i + dy j = "<<y00+dyx*i0+dy*j0<<std::endl;
+        int inti0 = int(floor(i0+0.5));
+        int intj0 = int(floor(j0+0.5));
+
+        if ( std::abs(i0 - inti0) < 1.e-12 && std::abs(j0 - intj0) < 1.e-12 &&
+             inti0 >= 0 && inti0 < m && intj0 >= 0 && intj0 < n)  {
```

Comment:
My guess is that this isn't likely to be worth doing.  For Sersic, it's important for high n Sersics where the central peak is so sharp that an error of 1.e-18 in the rsq value leads to an error of about 0.5% in the amplitude.  cf. [this comment](https://github.com/GalSim-developers/GalSim/issues/497#issuecomment-31847079).  

For Spergel, the really peaky ones (nu <= 0) actually have an infinite value at the center, so making this "fix" will put an infinity at that point, which isn't likely to be terribly useful.

In fact, this feature of Spergel profiles probably ought to be mentioned in the main doc string.  Namely, that for nu <= 0, the central value is undefined, so there could be problems rendering them directly without convolving by either a PSF or a pixel if the profile center is precisely on a pixel center.